bin/benchmark_performance.rb

```diff
@@ -0,0 +1,425 @@
+require 'tmpdir'
```

Comment:
This file is fairly long. Can it be broken down into modules and objects?

Reply:
Did a big refactoring into various files.